raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -306,10 +314,22 @@ private VoteResponseData buildVoteResponse(Errors error, boolean voteGranted) {
                 .setVoteGranted(voteGranted);
     }
 
-    private VoteResponseData handleVoteRequest(VoteRequestData request) throws IOException {
-        Optional<Exception> errorOpt = validateLeaderOnlyRequest(request.candidateId(), request.candidateEpoch());
+    /**
+     * Handle a Vote request. This API may return the following errors:
+     *
+     * - {@link Errors#BROKER_NOT_AVAILABLE} if this node is currently shutting down
+     * - {@link Errors#FENCED_LEADER_EPOCH} if the epoch is smaller than this node's epoch
+     * - {@link Errors#INCONSISTENT_VOTER_SET} if the request suggests inconsistent voter membership (e.g.
+     *      if this node or the sender is not one of the current known voters)
+     * - {@link Errors#INVALID_REQUEST} if the last epoch or offset are invalid
+     */
+    private VoteResponseData handleVoteRequest(
+        RaftRequest.Inbound requestMetadata
+    ) throws IOException {
+        VoteRequestData request = (VoteRequestData) requestMetadata.data;
+        Optional<Errors> errorOpt = validateVoterOnlyRequest(request.candidateId(), request.candidateEpoch());
         if (errorOpt.isPresent()) {
-            return buildVoteResponse(Errors.forException(errorOpt.get()), false);
+            return buildVoteResponse(errorOpt.get(), false);
```

Comment:
This is for line 336/340 below: I think these lines would only happen if the maintained epoch is exactly the same as the request epoch (if it is larger we'd return `FENCED_LEADER_EPOCH` in line 332, if it is smaller we'd no longer be leader / candidate any more). If that's the case then we could make the log4j to be more explicit like "... of epoch {} which is the same as the request epoch".